packages/datadog-plugin-fs/src/index.js

```diff
@@ -0,0 +1,497 @@
+    }
+    if (typeof fd !== 'number') {
+      return
+    }
+    return makeFSTags(fd, null, null, config, tracer)
+  }
+}
+
+function createChownTags (config, tracer) {
+  return function chownTags (path, uid, gid) {
+    if (typeof path === 'number' || typeof uid !== 'number' || typeof gid !== 'number') {
+      return
+    }
+    const tags = makeFSTags(path, null, null, config, tracer)
+    tags['file.uid'] = uid.toString()
+    tags['file.gid'] = gid.toString()
+    return tags
+  }
+}
+
+function createFchownTags (config, tracer) {
+  return function fchownTags (fd, uid, gid) {
+    if (typeof this === 'object' && this !== null && this.fd) {
+      fd = this.fd
+    }
+    if (typeof fd !== 'number' || typeof uid !== 'number' || typeof gid !== 'number') {
+      return
+    }
+    const tags = makeFSTags(fd, null, null, config, tracer)
+    tags['file.uid'] = uid.toString()
+    tags['file.gid'] = gid.toString()
+    return tags
+  }
+}
+
+function getSymbolName (sym) {
+  return sym.description || sym.toString()
+}
+
+const tagMakers = {
+  open: createOpenTags,
+  close: createCloseTags,
+  readFile: createReadFileTags,
+  writeFile: createWriteFileTags,
+  appendFile: createAppendFileTags,
+  access: createPathTags,
+  copyFile: createCopyFileTags,
+  stat: createPathTags,
+  lstat: createPathTags,
+  fstat: createFDTags,
+  readdir: createPathTags,
+  opendir: createPathTags,
+  read: createFDTags,
+  write: createFDTags,
+  writev: createFDTags,
+  chmod: createChmodTags,
+  lchmod: createChmodTags,
+  fchmod: createFchmodTags,
+  chown: createChownTags,
+  lchown: createChownTags,
+  fchown: createFchownTags,
+  realpath: createPathTags,
+  readlink: createPathTags,
+  unlink: createPathTags,
+  symlink: createCopyFileTags,
+  link: createCopyFileTags,
+  rmdir: createPathTags,
+  rename: createCopyFileTags,
+  fsync: createFDTags,
+  fdatasync: createFDTags,
+  mkdir: createPathTags,
+  truncate: createPathTags,
+  ftruncate: createFDTags,
+  utimes: createPathTags,
+  futimes: createFDTags,
+  mkdtemp: createPathTags
+}
+
+function createWrapCb (tracer, config, name, tagMaker) {
+  const makeTags = tagMaker(config, tracer)
+  name = 'fs.' + name
+  return function wrapFunction (fn) {
+    return function wrappedFunction () {
+      const cb = arguments[arguments.length - 1]
+      if (typeof cb !== 'function') {
+        return fn.apply(this, arguments)
+      }
+      const tags = makeTags.apply(this, arguments)
+      if (!tags) {
+        return fn.apply(this, arguments)
+      }
+      return tracer.trace(name, { tags }, (span, done) => {
+        arguments[arguments.length - 1] = wrapCallback(cb, done)
+        return fn.apply(this, arguments)
+      })
+    }
+  }
+}
+
+function createWrap (tracer, config, name, tagMaker) {
+  const makeTags = tagMaker(config, tracer)
+  name = 'fs.' + name
+  return function wrapSyncFunction (fn) {
+    return function wrappedSyncFunction () {
+      const tagArgs = Array.from(arguments)
```

Comment:
The copy is only used as the arguments to `makeTags`. Unshifting would work, but would change the arguments.

The issue here is that filehandle operations all take the same arguments as their `fs.f*()` counterparts, but omit the first (file descriptor) argument.

Looking at it again, i think this will work *without* modifying the args to makeTags, because the tagMaker here doesn't actually check if there's a file descriptor before checking if `this` is a filehandle. This whole chunk can be removed.